python/paddle/fluid/dygraph/dygraph_to_static/ifelse_transformer.py

```diff
@@ -132,31 +159,43 @@ class IsControlFlowIfVisitor(gast.NodeTransformer):
              because reshape_op may be called before this statement.
     """
 
-    def __init__(self, static_analysis_visitor):
-        self.static_analysis_visitor = static_analysis_visitor
+    def __init__(self, ast_node, static_analysis_visitor=None):
+        assert isinstance(
+            ast_node, gast.AST
+        ), "Type of input node should be gast.AST, but received %s." % type(
+            ast_node)
+        self.ast_root = ast_node
+        if static_analysis_visitor is None:
+            self.static_analysis_visitor = StaticAnalysisVisitor(ast_node)
+        else:
+            self.static_analysis_visitor = static_analysis_visitor
         self.node_to_wrapper_map = self.static_analysis_visitor.get_node_to_wrapper_map(
         )
-        self.is_control_flow = False
+        self.is_control_flow_num = 0
+        self._compare_node_tenor_set = set()
 
-    def transform(self, node):
-        if self._is_candidate_node(node):
+    def transform(self):
+        node = self.ast_root
+        if is_candidate_node(node):
             self.visit(node)
-        return self.is_control_flow
+        return self.is_control_flow_num > 0
 
     def visit_BoolOp(self, node):
-        for child in node.values:
-            if not self._is_candidate_node(child):
-                continue
-            self.generic_visit(node)
+        for i, child in enumerate(node.values):
+            if is_candidate_node(child):
+                self.visit(child)
         return node
 
     def visit_Compare(self, node):
         # Ignores child node with `if x` or `if x is None`
-        if not self._compare_with_none(node):
+        pre_control_flow_num = self.is_control_flow_num
```

Comment:
`if x` whill be different from `if x is None` after #22892 merged. A TODO comment can be added here.

Reply:
Thanks. Fixed.